packages/core/admin/utils/plugins.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+const asyncFs = require('fs/promises');
+const { camelCase } = require('lodash');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const glob = require('glob');
+
+/**
+ * @typedef {Object} PluginInfo
+ * @property {string} packageName
+ * @property {string} description
+ * @property {boolean=} required
+ */
+
+/**
+ * @typedef {Object} Plugin
+ * @property {string} pathToPlugin
+ * @property {string} name
+ * @property {PluginInfo} info
+ * @property {string=} directory
+ */
+
+/**
+ * @param {string[]} pluginsAllowlist
+ * @returns {Plugin[]}
+ */
+const getPlugins = (pluginsAllowlist) => {
+  const rootPath = path.resolve(__dirname, '..', path.join('..', '..', '..', 'packages'));
+  /**
+   * So `glob` only supports '/' as a path separator, so we need to replace
+   * the path separator for the current OS with '/'. e.g. on windows it's `\`.
+   *
+   * see https://github.com/isaacs/node-glob/#windows for more information
+   *
+   * and see https://github.com/isaacs/node-glob/issues/467#issuecomment-1114240501 for the recommended fix.
+   */
+  let corePath = path.join(rootPath, 'core', '*');
+  let pluginsPath = path.join(rootPath, 'plugins', '*');
+
+  if (process.platform === 'win32') {
+    corePath = corePath.split(path.sep).join(path.posix.sep);
+    pluginsPath = pluginsPath.split(path.sep).join(path.posix.sep);
+  }
+
+  const corePackageDirs = glob.sync(corePath);
+  const pluginsPackageDirs = glob.sync(pluginsPath);
+
+  const plugins = [...corePackageDirs, ...pluginsPackageDirs]
+    .map((directory) => {
+      const isCoreAdmin = directory.includes('packages/core/admin');
+
+      if (isCoreAdmin) {
+        return null;
+      }
+
+      const { name, strapi } = require(path.join(directory, 'package.json'));
+
+      /**
+       * this will remove any of our packages that are
+       * not actually plugins for the application
+       */
+      if (!strapi) {
+        return null;
+      }
```

Comment:
Could do, but then we're doing `filter->map->filter`, i think knowing this is being removed with the vite stuff i think it's fine, because we won't have these scripts in the admin package and we'll know via the strapi instance what plugins there actually are :)